python-spec/src/somacore/coordinates.py

```diff
@@ -0,0 +1,521 @@
+"""Definitions of types related to coordinate systems."""
+
+import abc
+import collections.abc
+from dataclasses import dataclass
+from typing import Any, Optional, Sequence, Tuple, Union
+
+import numpy as np
+import numpy.typing as npt
+
+
+@dataclass
+class Axis(metaclass=abc.ABCMeta):
+    """A description of an axis of a coordinate system
+
+    Args:
+        name: Name of the axis.
+        unit: Optional string units. Defaults to ``None``.
+        scale: Optional scale for units. Defaults to ``None``.
+
+    Lifecycle: experimental
+    """
+
+    name: str
+    units: Optional[str] = None
+    scale: Optional[np.float64] = None
```

Comment:
No, because the coordinates live apart from the transforms. Without the scale, it would be implied that the units are always factors of 1. But that's not necessary true.

For example, you can have a coordinate system where a single unit is "10 inches". I think it's nice to split the unit type and the scale of the unit into separate pieces. It also gives me the option of adjusting the scale if I applying a scale transformation to a coordinate space.

Reply:
> For example, you can have a coordinate system where a single unit is "10 inches".

I'm not sure I understand the use case for this, could you elaborate? If I did, I think I would define that as one axis whose unit was `"10 inches"` then have a scale transform to another coordinate system that was in inches.

Reply:
I renamed them to `unit_name` and `unit_scale`. Does that help clarify?

Reply:
I don't think the names are an issue.

Could you elaborate on use case for this? I'm having trouble understanding why you would want to have this when you can already define scaling with the transformations.

Reply:
The coordinate system exists outside of the transformation. The unit name and scale isn't describing an operation, it's telling you what the fixed size of a single unit for that axis is.

Reply:
Discussed synchronously. Adding the scale is non-standard and will make syncing with out specs/tools more difficult. If we want to use a coordinate system with non-standard units, we'll just provide the transform as well.